src/modules/navigator/takeoff.cpp

```diff
@@ -98,60 +98,31 @@ Takeoff::set_takeoff_position()
 {
 	struct position_setpoint_triplet_s *rep = _navigator->get_takeoff_triplet();
 
-	float abs_altitude = 0.0f;
+	float takeoff_altitude_amsl = 0.f;
 
-	float min_abs_altitude;
-
-	// TODO: review this, comments are talking about home pos, the validity is checked but the
-	// current altitude is used instead. Also, the "else" case does not consider the current altitude at all.
-	if (_navigator->home_alt_valid()) { //only use home position if it is valid
-		min_abs_altitude = _navigator->get_global_position()->alt + _navigator->get_takeoff_min_alt();
-
-	} else { //e.g. flow
-		min_abs_altitude = _navigator->get_takeoff_min_alt();
-	}
-
-	// Use altitude if it has been set. If home position is invalid use min_abs_altitude
-	events::LogLevel log_level = events::LogLevel::Disabled;
-
-	if (rep->current.valid && PX4_ISFINITE(rep->current.alt) && _navigator->home_alt_valid()) {
-		abs_altitude = rep->current.alt;
-
-		// If the altitude suggestion is lower than home + minimum clearance, raise it and complain.
-		if (abs_altitude < min_abs_altitude) {
-			if (abs_altitude < min_abs_altitude - 0.1f) { // don't complain if difference is smaller than 10cm
-				mavlink_log_critical(_navigator->get_mavlink_log_pub(),
-						     "Using minimum takeoff altitude: %.2f m\t", (double)_navigator->get_takeoff_min_alt());
-				log_level = events::LogLevel::Warning;
-			}
-
-			abs_altitude = min_abs_altitude;
-		}
+	if (rep->current.valid && PX4_ISFINITE(rep->current.alt)) {
+		takeoff_altitude_amsl = rep->current.alt;
```

Comment:
Can we always assume that the absolute altitude is stored for the triplets? Why not using the get_absolute_altitude_for_item function?

Reply:
It's always absolute, see https://github.com/PX4/PX4-Autopilot/blob/6bd13c551498d9701ffc92f88e9dbf1425820220/msg/PositionSetpoint.msg#L21.

The special handling in Mission is required as the mission item altitude can indeed be relative.